salt/roster/ansible.py

```diff
@@ -113,189 +105,46 @@
-        if varname not in self.hostvars:
-            self.hostvars[varname] = dict()
-        self.hostvars[varname][key] = value
-
-    def _parse_parents_line(self, line, varname):
-        '''
-        Parse lines in the inventory file that are under the same [*:children] block
-        '''
-        if varname not in self.parents:
-            self.parents[varname] = []
-        self.parents[varname].append(line)
-
-
-class Script(Target):
+def targets(tgt, tgt_type='glob', **kwargs):
     '''
-    Matcher for Inventory scripts
+    Return the targets from the ansible inventory_file
+    Default: /etc/salt/roster
     '''
-    def __init__(self, tgt, tgt_type='glob', inventory_file='/etc/salt/roster'):
-        self.tgt = tgt
-        self.tgt_type = tgt_type
-        inventory, error = subprocess.Popen([inventory_file], shell=True, stdout=subprocess.PIPE).communicate()
-        self.inventory = salt.utils.json.loads(salt.utils.stringutils.to_str(inventory))
-        self.meta = self.inventory.get('_meta', {})
-        self.groups = dict()
-        self.hostvars = dict()
-        self.parents = dict()
-        for key, value in six.iteritems(self.inventory):
-            if key == '_meta':
-                continue
-            if type(value) is list:
-                self._parse_groups(key, value)
-            if 'hosts' in value:
-                self._parse_groups(key, value['hosts'])
-            if 'children' in value:
-                self._parse_parents(key, value['children'])
-            if 'hostvars' in value:
-                self._parse_hostvars(key, value['hostvars'])
-
-    def _parse_groups(self, key, value):
-        '''
-        Parse group data from inventory_file
-        '''
-        host = dict()
-        if key not in self.groups:
-            self.groups[key] = dict()
-        for server in value:
-            tmp = self.meta.get('hostvars', {}).get(server, False)
-            if tmp is not False:
-                if server not in host:
-                    host[server] = dict()
-                for tmpkey, tmpval in six.iteritems(tmp):
-                    if tmpkey in CONVERSION:
-                        host[server][CONVERSION[tmpkey]] = tmpval
-                if 'sudo' in host[server]:
-                    host[server]['passwd'], host[server]['sudo'] = host[server]['sudo'], True
-
-        self.groups[key].update(host)
-
-    def _parse_hostvars(self, key, value):
-        '''
-        Parse hostvars data from inventory_file
-        '''
-        if key not in self.hostvars:
-            self.hostvars[key] = dict()
-        self.hostvars[key] = value
-
-    def _parse_parents(self, key, value):
-        '''
-        Parse children data from inventory_file
-        '''
-        if key not in self.parents:
-            self.parents[key] = []
-        self.parents[key].extend(value)
+    inventory = __runner__['salt.cmd']('cmd.run', 'ansible-inventory -i {0} --list'.format(get_roster_file(__opts__)))
```

Comment:
I am really wondering why we would execute that in so heavy-weight fashion spinning off another Python interpreter etc, instead of just reuse the `ansible.cli.InventoryCLI` for this. As an example:

```python
import ansible.cli.inventory

c = ansible.cli.inventory.InventoryCLI(['', '-i', get_roster_file(__opts__), '--list'])
c.parse()
inventory = c.run()
```

Reply:
We still support python 3.4, which ansible does not, so in order to use ansible-inventory on centos 7, right now you have to pip2 install ansible, and then run the binary.

Until we drop python3.4 support i think this is the better setup.

Reply:
Ah, you're right. CentOS 7 and family on 3.4. OK, in this case we should do the fallback then:

```
return (ansible_inventory or which('ansible-inventory')) and __virtualname__, '....'
```

I still don't want to spin-off another Python process on an expensive IBM Mainframe. :-)

Reply:
```
.tox/py36/lib/python3.6/site-packages/ansible/cli/inventory.py:186: in run
    exit(0)
_ _ _ _ _ _ _ _
```

This is why we cannot use the ansible/cli/inventory.py script, because it calls a system exit.